qiskit/_quantumprogram.py

```diff
@@ -909,23 +908,32 @@ def compile(self, name_of_circuits, backend="local_qasm_simulator",
         qobj["config"] = {"max_credits": max_credits, 'backend': backend,
                           "shots": shots}
         qobj["circuits"] = []
-
+        backend_conf = qiskit.backends.get_backend_configuration(backend)
+        if not basis_gates:
+            if 'basis_gates' in backend_conf:
+                basis_gates = backend_conf['basis_gates']
```

Comment:
Would it make sense to preserve the previous behaviour of setting `basis_gates = "u1,u2,u3,cx,id"` (old line 921) explicitly if `basis_gates` is not provided by the function argument nor the backend? Something similar to:
```python
if not basis_gates and 'basis_gates' in backend_conf:
    basis_gates = backend_conf['basis_gates']
else:
    basis_gates = "u1,u2,u3,cx,id"  # QE target basis
```

As far as I can see, if `basis_gates == None` after these lines, it should behave the same way as if it was `basis_gates == "u1,u2,u3,cx,id"` (as eventually it gets set to the same value [here](https://github.com/ewinston/qiskit-sdk-py/blob/c2d860d815a27d6c6ea831d5fb7f3935d94c52cb/qiskit/_openquantumcompiler.py#L109)) - but making it explicit would help a bit when we get to revise the `basis_gates` handling as a whole.

Reply:
I think that may have removed it from _quantumprogram because it seemed like the "default" was defined in too many places. It thought maybe it would be easier to understand the code if the default was only defined just before it was needed i.e. the "compiler"?